util/io/src/worker.rs

```diff
@@ -69,37 +71,39 @@ impl Worker {
 		worker.thread = Some(thread::Builder::new().stack_size(STACK_SIZE).name(format!("IO Worker #{}", index)).spawn(
 			move || {
 				LOCAL_STACK_SIZE.with(|val| val.set(STACK_SIZE));
-				Worker::work_loop(stealer, channel.clone(), wait, wait_mutex.clone(), deleting)
+				let mut runtime = match tokio::runtime::current_thread::Runtime::new() {
+					Ok(c) => c,
+					Err(e) => {
+						error!(target: "ioworker", "error while executing future: {}", e);
```

Comment:
Would be good change this error message to distinguish between when the actual `future` fails and when `Tokio::Runtime::CurrentThread` fails.

So I suggest changing this error message